### PR TITLE
Downlod replies

### LINKED ISSUE #108: Add reply downloading, decryption, and display in conversation view
We currently have a placeholder for the display of replies in the conversation view. Now that message download and decryption has landed in master, we should extend this to also download and decrypt replies, and replace the placeholder in the conversation view with the content of the reply.

### PR DESCRIPTION
**Pull request by joshuathayer**
Building off #99, also spawn a thread for downloading and processing Replies from the journalist.

Aside from some minor refactoring to pull out common code between the Message the Reply fetching threads, there's not much novel here. It follows the Message-fetching model pretty closely.

Closes #108

### CODE REVIEW

File: securedrop_client/storage.py [hunk 1]
```diff
@@ -246,6 +246,13 @@ def find_new_submissions(session):
     return submissions
 
 
+def find_new_replies(session):
+    submissions = session.query(Reply) \
```
Comment (joshuathayer) [inline securedrop_client/storage.py]: `s/submissions/replies`

File: securedrop_client/storage.py [hunk 2]
```diff
@@ -256,3 +263,15 @@ def mark_file_as_downloaded(uuid, session):
     submission_db_object.is_downloaded = True
     session.add(submission_db_object)
     session.commit()
+
+
+def mark_reply_as_downloaded(uuid, session):
+    """Mark reply as downloaded in the database. The file itself will be
+    stored in the data directory.
+    """
+    submission_db_object = session.query(Reply) \
```
Comment (joshuathayer) [inline securedrop_client/storage.py]: same, `s/submission/reply`

**Comment by redshiftzero** [comment]
@joshuathayer and I reviewed this in person, looks good! Will test in Qubes and plan to merge this in for inclusion in the first `securedrop-client` deb

(To the interested observer, the advantage of having these separate reply and message threads is parallel download/decrypt happening for replies and messages, cc: #106)

**Comment by redshiftzero** [comment]
hey so I was on this branch and I hit this error:

```
Running app with home directory: /var/folders/gv/2r10zvfj303fyd5_6yn_p0b80000gn/T/tmp.EEN4XTsi
Traceback (most recent call last):
  File "/Users/redshiftzero/Documents/Github/securedrop-client/securedrop_client/gui/main.py", line 158, in on_source_changed
    self.show_conversation_for(self.current_source)
  File "/Users/redshiftzero/Documents/Github/securedrop-client/securedrop_client/gui/main.py", line 189, in show_conversation_for
    "<Reply not yet downloaded>")
  File "/Users/redshiftzero/Documents/Github/securedrop-client/securedrop_client/gui/main.py", line 167, in add_item_content_or
    adder(item.content)
  File "/Users/redshiftzero/Documents/Github/securedrop-client/securedrop_client/models.py", line 24, in content
    return self.data.get(fn_no_ext)
  File "/Users/redshiftzero/Documents/Github/securedrop-client/securedrop_client/data.py", line 16, in get
    msg = fh.read()
  File "/Users/redshiftzero/.virtualenvs/securedrop-client-l0Id-Ywn/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xca in position 0: unexpected end of data
```

The decrypted replies are getting written to the data directory but they are only one byte long.. I have to run but I will work on debugging this first thing tomorrow (if someone else notices the issue, please comment!)

**Comment by redshiftzero** [comment]
figured it out, it was indeed non-Qubes only due to the development test data: https://github.com/freedomofpress/securedrop/pull/3927 resolves

**Pull request merged**